docs/operate/get-started/other-hardware/_index.md

```diff
@@ -51,48 +51,69 @@ For C++ module examples, see the [C++ examples directory on GitHub](https://gith
 {{% /alert %}}
 
 {{< expand "How and where do modules run?" >}}
-Modules run on your machine, alongside [`viam-server`](/operate/reference/viam-server/) as separate processes, communicating with `viam-server` over UNIX sockets.
-
-When a module initializes, it registers its {{< glossary_tooltip term_id="model" text="model or models" >}} and associated [APIs](/dev/reference/apis/) with `viam-server`, making the new model available for use.
-`viam-server` manages the [dependencies](/operate/reference/viam-server/#dependency-management), [start-up](/operate/reference/viam-server/#start-up), [reconfiguration](/operate/reference/viam-server/#reconfiguration), [data management](/data-ai/capture-data/capture-sync/), and [shutdown](/operate/reference/viam-server/#shutdown) behavior of your modular resource.
+Modules run on your machine, alongside `viam-server` as separate processes, communicating with `viam-server` over UNIX sockets.
+
+[`viam-server` manages](/operate/reference/viam-server/) the dependencies, start-up, reconfiguration, [data management](/data-ai/capture-data/capture-sync/), and shutdown behavior of your modular resource.
+
+The lifecycle of a module and the resources it provides is as follows:
+
+1. `viam-server` starts, and if it is connected to the internet, it checks for configuration updates.
+1. `viam-server` starts any configured modules.
+1. When a module initializes, it registers its model or models and associated [APIs](/dev/reference/apis/) with `viam-server`, making the models available for use.
+1. For each modular resource configured on the machine, `viam-server` uses the resource's `validate` function and the `depends_on` field in the resource configuration to determine the dependencies of the resource.
+1. If a dependency is not already running, `viam-server` starts it before starting the resource.
+   If a dependency is not found or fails to start, `viam-server` will not start the resource that depends on it.
+1. `viam-server` calls the resource's constructor to build the resource based on its configuration.
+   Typically, the constructor calls the reconfigure function.
+1. If configuration fails due to a validation failure or an exception thrown by the reconfigure or constructor functions, `viam-server` attempts to reconfigure the resource.
+1. The modular resource is ready to use.
```

Comment:
This should also cover the failure scenario.  How often does it retry? What happens if it fails? Otherwise it's a bit odd to go from here to the next step. 
What makes it ready to use? If reconfigure succeeds? Can we be more specific?

Reply:
@benjirewis If reconfiguration happens within 15 seconds, does that also mean that initial configuration is retried every 15 seconds until it times out?
```suggestion
1. If configuration fails due to a validation failure or an exception thrown by the reconfigure or constructor functions, `viam-server` attempts to reconfigure the resource every 15 seconds.
   If the module exceeds the [configured timeout limits](/operate/reference/module-configuration/#environment-variables) (default 5 minutes to start up and 1 minute to reconfigure), `viam-server` logs an [error](/dev/tools/common-errors/#timed-out-waiting-for-module).
1. Once the modular resource has started up and configured, it is available for use.
```

Reply:
mentioning the env variables is a good idea

Reply:
> @benjirewis If reconfiguration happens within 15 seconds, does that also mean that initial configuration is retried every 15 seconds until it times out?

So we have a ticker that checks for "unconfigured" resources (never constructed) every _5_ seconds. We'll try to construct a not-yet-successfully-constructed modular resource every _5_ seconds as long as `viam-server` and the module are alive and the resource is not successfully constructing. Same thing goes for reconfigure and a not-yet-successfully-reconfigured modular resource.

Reply:
Am I erroneously conflating construction and reconfiguration? I've thought of them as conjoined since the constructor calls reconfigure right? Should the wording of the suggestion above change @benjirewis ?

Reply:
Yeah if you want to be super pedantic it would be:
```
If construction fails due to a validation failure or an exception thrown by the modular resource's constructor, `viam-server` attempts to construct the resource every 5 seconds.

Similarly, if reconfiguration fails due to a validation failure or an exception thrown by the modular resource's reconfigure method, `viam-server` attempts to reconfigure the resource every 5 seconds.
```